packages/app-cli/app/command-mv.js

```diff
@@ -7,25 +7,45 @@ const Note = require('@joplin/lib/models/Note').default;
 
 class Command extends BaseCommand {
 	usage() {
-		return 'mv <note> [notebook]';
+		return 'mv <item> [notebook]';
 	}
 
 	description() {
-		return _('Moves the notes matching <note> to [notebook].');
+		return _('Moves the given <item> (notes matching pattern in current notebook or one notebook) to [notebook]. If <item> is subnotebook and [notebook] is "root", will make <item> parent notebook');
 	}
 
 	async action(args) {
-		const pattern = args['note'];
+		const pattern = args['item'];
 		const destination = args['notebook'];
+		let folder = null;
 
-		const folder = await Folder.loadByField('title', destination);
-		if (!folder) throw new Error(_('Cannot find "%s".', destination));
+		if (destination != 'root') {
+			folder = await app().loadItem(BaseModel.TYPE_FOLDER, destination);
+			if (!folder) throw new Error(_('Cannot find "%s".', destination));
+		}
 
-		const notes = await app().loadItems(BaseModel.TYPE_NOTE, pattern);
-		if (!notes.length) throw new Error(_('Cannot find "%s".', pattern));
+		const dstDups = await Folder.search({ titlePattern: destination, limit: 2 });
+		if (dstDups.length > 1) {
+			throw new Error(_('Ambiguous notebook "%s". Please use notebook id instead - look for parent id in metadata', destination));
+		}
 
-		for (let i = 0; i < notes.length; i++) {
-			await Note.moveToFolder(notes[i].id, folder.id);
+		const itemFolder = await app().loadItem(BaseModel.TYPE_FOLDER, pattern);
+		if (itemFolder) {
+			const srcDups = await Folder.search({ titlePattern: pattern, limit: 2 });
+			if (srcDups.length > 1) {
+				throw new Error(_('Ambiguous notebook "%s". Please use notebook id instead - look for parent id in metadata or use $b for current', pattern));
+			}
+			if (destination == 'root') {
+				await Folder.moveToFolder(itemFolder.id, '');
+			} else {
+				await Folder.moveToFolder(itemFolder.id, folder.id);
+			}
+		} else {
+			const notes = await app().loadItems(BaseModel.TYPE_NOTE, pattern);
+			if (!notes.length) throw new Error(_('Cannot find "%s".', pattern));
+			for (let i = 0; i < notes.length; i++) {
+				await Note.moveToFolder(notes[i].id, folder.id);
+			}
```

Comment:
```suggestion
			await Promise.all(notes.map(note => Note.moveToFolder(note.id, folder.id)));
```
The above suggestion **might be a bad idea** — while [`Promise.all`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/all) allows all `Note.moveToFolder` to run concurrently (and thus possibly be faster), I don't know if this will break `moveToFolder`...

Reply:
I have tested it with 136 notes (`mv * shortid`) and I personally like the suggestion it is actually faster. However, I would also wait for the feedback of the others.

Reply:
@laurent22 What do you think about it?

Reply:
It think it's better not to Promise.all here I think. What if the user tries to move 10,000 notes. Will it try to move them all at the same time?

Reply:
I'd say if the user moved 10,000 notes they shouldn't be surprised if the app becomes unresponsive for some time. Maybe it's better to block while moving, than moving notes in background while the user can continue editing.

Reply:
Unresponsive maybe, but maybe it will get stuck or crash? If it's in a loop the workload is predictable, it just takes time. But everything in parallel, it's harder to know what will happen and it might also depend on hardware or OS.

Reply:
Thanks for your opinion, i would currently stay on the current loop implementation.

My opinion, if someone have performance issues with move they can open an additional Issue. My knowledge is unfortunately very minimal and would take more time as i currently want invest in this issue.